dask_kubernetes/core.py

```diff
@@ -425,33 +495,8 @@ def __init__(
             self.env if self.env is not None else dask.config.get("kubernetes.env")
         )
 
-        if not self.pod_template and dask.config.get(
-            "kubernetes.worker-template", None
-        ):
-            d = dask.config.get("kubernetes.worker-template")
-            d = dask.config.expand_environment_variables(d)
-            self.pod_template = make_pod_from_dict(d)
-
-        if not self.pod_template and dask.config.get(
-            "kubernetes.worker-template-path", None
-        ):
-            import yaml
-
-            fn = dask.config.get("kubernetes.worker-template-path")
-            fn = fn.format(**os.environ)
-            with open(fn) as f:
-                d = yaml.safe_load(f)
-            d = dask.config.expand_environment_variables(d)
-            self.pod_template = make_pod_from_dict(d)
-
-        if not self.pod_template:
-            msg = (
-                "Worker pod specification not provided. See KubeCluster "
-                "docstring for ways to specify workers"
-            )
-            raise ValueError(msg)
+        self._clean_pod_templates()
```

Comment:
This function is a bit **magic**. It's not obvious what the result of calling it is. I would prefer a direct assignment for clarity.